packages/nuxt/src/app/nuxt.ts

```diff
@@ -23,7 +23,7 @@ type NuxtMeta = {
 
 type HookResult = Promise<void> | void
 
-type AppRenderedContext = { ssrContext: NuxtApp['ssrContext'] }
+type AppRenderedContext = { ssrContext: NuxtApp['ssrContext'], renderResult: Awaited<ReturnType<ReturnType<typeof createRenderer>['renderToString']>> }
```

Comment:
We might explicitly type both renderResult and ssrContext as readonly since changes are not applied